contracts/src/periphery/cert/router/EigenDACertVerifierRouter.sol

```diff
@@ -41,7 +42,9 @@ contract EigenDACertVerifierRouter is IEigenDACertVerifierRouter, OwnableUpgrade
     }
 
     function addCertVerifier(uint32 activationBlockNumber, address certVerifier) external onlyOwner {
-        if (activationBlockNumber < block.number) {
+        // We disallow adding cert verifiers at the current block number to avoid a race condition of
+        // adding a cert verifier at the current block and verifying in the same block
+        if (activationBlockNumber <= block.number) {
```

Comment:
this is actually a great catch! We want cert verification at a block number to be deterministic. That can only be the case if block numbers are activated as a precursor to a given block's execution, which is exactly what our current design does now.

Otherwise you can have in a single block:
1. tx1 verifies against certVerifier N
2. certVerifier N+1 gets activated
3. tx2 verifies same cert as tx1 but against certVerifier N+1

Reply:
Right! I think these checks really need to be specified up front in a spec because it is hard to know whether this is even an invariant I want to protect without that. We should not just add require statements ad hoc because we can miss interactions across different functions.